nebu/cli/get.py

```
@@ -158,88 +182,189 @@ def store_sha1(sha1, write_dir, filename):
-        """
-        # skip index.cnxml.html
-        resources = {res['filename']: res for res in metadata['resources']
-                     if res['filename'] != 'index.cnxml.html'}
-
-        # Deal with core XML file and output directory
-        filename = filename_by_type[metadata['mediaType']]
-        url = '{}/resources/{}'.format(base_url, resources[filename]['id'])
-        file_resp = requests.get(url)
-        if not(book_tree) and filename == 'index.cnxml':
-            write_dir = write_dir / metadata['legacy_id']
-            os.mkdir(str(write_dir))
+async def do_with_retried_request_while(session,
+                                        url,
+                                        condition,
+                                        max_retries,
+                                        do,
+                                        sleep_time=0.5):
+    retries = 0
+    while retries <= max_retries:
+        try:
+            async with session.get(
+                    url,
+                    timeout=ClientTimeout(total=15)) as response:
+                if not condition(response):
+                    return await do(response)
+        except asyncio.TimeoutError:  # pragma: no cover
+            pass
+        retries += 1
+        await asyncio.sleep(sleep_time)
+    raise Exception(f'Max retries exceeded: {url}')
+
+
+async def _write_contents(tree,
+                          base_url,
+                          out_dir,
+                          book_tree=False,
+                          get_resources=False,
+                          pbar=None):
+    async def fetch_content_meta_node(session,
+                                      node,
+                                      content_meta_url,
+                                      write_dir,
+                                      index_in_group=0):
+        async def get_metadata():
+            async def response_json(response):
+                if response.status >= 400:
```

Can this be more precise? It seems that this will also silently continue when the response is a 5xx

---

Doh! I missed that one!